pkg/dynamiclifecycle/manager.go

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dynamiclifecycle
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/rate"
+	"github.com/cilium/cilium/pkg/time"
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/reconciler"
+
+	"github.com/cilium/cilium/pkg/dynamicconfig"
+)
+
+type FeatureStatus struct {
+	Feature DynamicFeature
+	Enabled bool
+}
+
+type Manager interface {
+	AddFeature(DynamicFeature, ...DynamicFeature)
+	Append(DynamicFeature, cell.HookInterface)
+	Start(DynamicFeature)
+	Stop(DynamicFeature)
+
+	initDynamicConfigWatcher()
+}
+
+type DefaultManager struct {
+	l   *slog.Logger
+	tl  statedb.RWTable[TableLifecycle]
+	dct statedb.Table[dynamicconfig.DynamicConfig]
+	db  *statedb.DB
+	jg  job.Group
+	mc  config
+	rt  *rate.Limiter
+}
+
+// AddFeature adds a feature to the TableLifecycle including the hard dependencies
+// The Feature name and the dependencies are immutable.
+// The dependencies are a list of DynamicFeature which need to be running before
+// this specific feature can be enabled.
+// If the dependencies are not enabled the DynamicFeature will keep retrying until all dependencies are enabled.
+func (d *DefaultManager) AddFeature(feature DynamicFeature, deps ...DynamicFeature) {
+	_, _, found := d.tl.Get(d.db.ReadTxn(), ByFeature(feature))
+	if found {
+		d.l.Error("DynamicFeature is already added", "feature", feature)
+		return
+	}
+
+	writeTxn := d.db.WriteTxn(d.tl)
+	defer writeTxn.Commit()
+
+	fl := newFeatureLifecycle(feature, deps...)
+	tl := newTableLifecycle(fl)
+	_, _, _ = d.tl.Insert(writeTxn, tl)
+
+	d.l.Info("DynamicFeature added", "feature", feature, "deps", deps)
+}
+
+// Start enables the DynamicFeature.
+// The Lifecycle hooks are started by the TableLifecycle reconciler considering the dependencies.
+// If one of the DynamicFeature hooks fail to start, the reconciler attempts
+// to stop already started hooks. The reconciler will retry to start the DynamicFeature.
+func (d *DefaultManager) Start(feature DynamicFeature) {
```

Comment:
Since this just enables it, perhaps we could just have `SetEnabled(feature DynamicFeature, enabled bool)` instead of start and stop?